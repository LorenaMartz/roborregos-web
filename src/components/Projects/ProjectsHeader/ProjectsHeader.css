.projects-header {
  height: 100%;
  position: relative;
  background: linear-gradient( 45deg, rgba(20, 18, 19, 0.9) 10%, rgba(0, 25, 105, 0.9) 100%), url(../../../images/github_pattern.svg) top left/auto repeat fixed;
  color: rgba(255, 255, 255, 0.8);
}

.projects-header .main-text-projects {
  color: rgba(255, 255, 255, 0.9);
  font-size: 3vh;
  text-transform: none;
  padding-top: 2%;
  padding-bottom: 0%;
}

.title-text-banner {
  color: rgba(255, 255, 255, 0.9);
  font-size: 7vh;
  text-transform: none;
}

.projects-header .projects-header-footer {
  font-size: 2.5vh;
  text-transform: none;
  display: flex;
  flex-direction: column;
  align-items: center;
  text-align: center;
  position: absolute;
  bottom: 5vh;
  left: 0;
  right: 0;
  margin-left:auto;
  margin-right:auto;
}

.projects-header .projects-header-footer p {
  margin: 0;
}

.projects-header .projects-header-footer .icon-btn {
  cursor: pointer;
  color: rgb(0, 225, 250);
  font-size: 30px;
}

.projects-header .projects-header-footer .icon-btn:hover {
  color: rgb(196, 63, 101);
}
