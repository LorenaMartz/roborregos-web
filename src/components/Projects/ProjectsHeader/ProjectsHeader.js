// @flow
import React, { Component } from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faAngleDown } from '@fortawesome/free-solid-svg-icons'
import './ProjectsHeader.css'

type Props = {};

/** Component class of Projects' header. */
class ProjectsHeader extends Component<Props> {
  /**
 Function to scroll down the window view towards th end of the component.
 */
  scrollToInfo = () => {
    window.scrollBy(0, window.innerHeight - 48 - window.scrollY)
  }

  /**
 * Renders Responsive view of Projects's header.
 * @return {renderized_component} Heder banner with legend.
 */
  render() {
    return (
      <div className="projects-header">
        <div className="container-legend">
          <h2 className="title-text-banner">
            Projects
          </h2>
          <div className="main-text-projects">
            <p>
              Every year we challenge ourselves to keep learning and improve.
              See what we&apos;ve been up to!
            </p>
          </div>
        </div>
        <div className="projects-header-footer">
          <p>
            Scroll down and meet the us!
          </p>
          <FontAwesomeIcon
            onClick={this.scrollToInfo}
            icon={faAngleDown}
            className="icon-btn"
          />
        </div>
      </div>
    )
  }
}

export default ProjectsHeader
