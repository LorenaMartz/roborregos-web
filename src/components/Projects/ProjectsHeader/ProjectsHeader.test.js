// @flow
import React from 'react'
import renderer from 'react-test-renderer'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import ProjectsHeader from './ProjectsHeader'

let wrapper
configure({ adapter: new Adapter() })

beforeEach(() => {
  wrapper = shallow(
    <ProjectsHeader />,
  )
})

it('<ProjectsHeader> renders components correctly', () => {
  expect(wrapper.find('.container-legend').length).toBe(1)
  expect(wrapper.find('.container-legend').length).toBe(1)
  expect(wrapper.find('.main-text-projects').length).toBe(1)
  expect(wrapper.find('.projects-header-footer').length).toBe(1)
  expect(wrapper.find('.icon-btn').length).toBe(1)
})

it('<ProjectsHeader> scrolls when "Learn More" button is clicked', () => {
  global.scrollBy = jest.fn()
  wrapper.find('.icon-btn').simulate('click')
  expect(global.scrollBy).toHaveBeenCalledWith(0, global.innerHeight - 48 - global.scrollY)
})

// Snapshots
it('<ProjectsHeader> matches snapshot', () => {
  const tree = renderer.create(
    <ProjectsHeader />,
  ).toJSON()
  expect(tree).toMatchSnapshot('ProjectsHeaderView.test.js.snap')
})
