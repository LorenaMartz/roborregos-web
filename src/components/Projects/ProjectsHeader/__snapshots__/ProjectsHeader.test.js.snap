// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectsHeader> matches snapshot: ProjectsHeaderView.test.js.snap 1`] = `
<div
  className="projects-header"
>
  <div
    className="container-legend"
  >
    <h2
      className="title-text-banner"
    >
      Projects
    </h2>
    <div
      className="main-text-projects"
    >
      <p>
        Every year we challenge ourselves to keep learning and improving.
         
        <br />
        See what we've been up to!
      </p>
    </div>
  </div>
  <div
    className="projects-header-footer"
  >
    <p>
      Learn more
    </p>
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-angle-down fa-w-10 icon-btn"
      data-icon="angle-down"
      data-prefix="fas"
      focusable="false"
      onClick={[Function]}
      role="img"
      style={Object {}}
      viewBox="0 0 320 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;
